lib/location/cellular/cellular_service.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef CELLULAR_SERVICE_H_
+#define CELLULAR_SERVICE_H_
+
+#include <modem/location.h>
+#include <modem/lte_lc.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Cellular positioning input parameters. */
+struct location_cellular_serv_pos_req {
+	/** Cellular positioning service to be used. */
+	enum location_service service;
+	/** Neighbor cell data. */
+	const struct lte_lc_cells_info *cell_data;
+	/**
+	 * @brief Timeout (in milliseconds) of how long the cellular positioning procedure can take.
+	 * SYS_FOREVER_MS means that the timer is disabled.
+	 */
+	int32_t timeout;
+};
+
+/**
+ * @brief Provision TLS certificate that the selected location service requires
+ *	  for HTTPS connections.
+ *	  Certificate provisioning must be done before location requests can
+ *	  successfully be executed, either using this API or some other method.
+ *
+ * @note Certificate provisioning must happen when LTE is not active in the modem.
+ *	 This is typically achieved by calling this API prior to setting up the
+ *	 LTE link in an application, but it can also be done by setting the
+ *	 modem in offline mode before provisioning.
+ *
+ * @param[in] overwrite If this flag is set, any CA certificate currently
+ *		        provisioned to CONFIG_LOCATION_METHOD_CELLULAR_TLS_SEC_TAG is
+ *		        overwritten.
+ *
+ * @return 0 on success, or negative error code on failure.
+ */
+int cellular_service_provision_certificate(bool overwrite);
```

Comment:
I'm not sure where should we put this but cellular side doesn't sound like correct place although I'm not willing to refactor too much with this PR.

Location utils, or a new location_utils_cert.h?

Reply:
yes location_utils sounds like a correct place and this should be called also from wifi side for nrf cloud.

Reply:
Yep, location_utils was on my mind exactly because this could be used from Wi-Fi side too.

Reply:
There is a separate commit now that moves certificate handling to location_service_utils.h/c. Please check.